src/human-readable/types/utils.ts

```diff
@@ -341,14 +375,22 @@ T extends `(${infer Parameters})`
 // Split name and modifier (e.g. `indexed foo` => `{ name: 'foo', indexed: true }`)
 export type _SplitNameOrModifier<
   T extends string,
-  Options extends ParseOptions = DefaultParseOptions,
+  Options extends ParseOptions & { type?: string } = DefaultParseOptions,
 > = Trim<T> extends infer Trimmed
   ? Options extends { Modifier: Modifier }
-    ? // TODO: Check that modifier is allowed
-      Trimmed extends `${infer Mod extends Options['Modifier']} ${infer Name}`
-      ? { readonly name: Trim<Name> } & (Mod extends 'indexed'
-          ? { readonly indexed: true }
-          : object)
+    ? Trimmed extends `${infer Mod extends Options['Modifier']} ${infer Name}`
+      ? Options extends { type: string; Structs: StructLookup | unknown }
+        ? ValidateModifier<Mod, Options> extends infer Validated
+          ? Validated extends string[]
+            ? { readonly name: Validated }
```

Comment:
Returns the string[] so that later when the parameters gets validated it keeps the error message

Reply:
Now returns the `string` instead of `string[]` so it get merged into to the parameter.